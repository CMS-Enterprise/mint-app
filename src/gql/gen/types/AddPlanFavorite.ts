/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: AddPlanFavorite
// ====================================================

export interface AddPlanFavorite_addPlanFavorite {
  __typename: "PlanFavorite";
  modelPlanID: UUID;
  userID: UUID;
}

export interface AddPlanFavorite {
  addPlanFavorite: AddPlanFavorite_addPlanFavorite;
}

export interface AddPlanFavoriteVariables {
  modelPlanID: UUID;
}
