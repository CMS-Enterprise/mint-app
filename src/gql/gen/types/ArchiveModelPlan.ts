/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: ArchiveModelPlan
// ====================================================

export interface ArchiveModelPlan_updateModelPlan {
  __typename: "ModelPlan";
  id: UUID;
}

export interface ArchiveModelPlan {
  updateModelPlan: ArchiveModelPlan_updateModelPlan;
}

export interface ArchiveModelPlanVariables {
  id: UUID;
  archived: boolean;
}
