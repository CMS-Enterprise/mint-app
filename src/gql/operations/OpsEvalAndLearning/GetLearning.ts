import { gql } from '@apollo/client';

export default gql(/* GraphQL */ `
  query GetLearning($id: UUID!) {
    modelPlan(id: $id) {
      id
      modelName
      opsEvalAndLearning {
        id
        ccmInvolvment
        dataNeededForMonitoring
        iddocSupport
        modelLearningSystems
        modelLearningSystemsOther
        modelLearningSystemsNote
        anticipatedChallenges
        readyForReviewByUserAccount {
          id
          commonName
        }
        readyForReviewDts
        status
      }
    }
  }
`);
