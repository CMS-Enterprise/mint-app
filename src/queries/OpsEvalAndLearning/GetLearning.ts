import { gql } from '@apollo/client';

export default gql`
  query GetLearning($id: UUID!) {
    modelPlan(id: $id) {
      id
      modelName
      opsEvalAndLearning {
        id
        ccmInvolvment
        dataNeededForMonitoring
        iddocSupport
        modelLearningSystems
        modelLearningSystemsOther
        modelLearningSystemsNote
        anticipatedChallenges
        readyForReviewBy
        readyForReviewDts
        status
      }
      operationalNeeds {
        modifiedDts
      }
    }
  }
`;
