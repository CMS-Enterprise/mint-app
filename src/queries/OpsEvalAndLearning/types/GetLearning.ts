/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { CcmInvolvmentType, DataForMonitoringType, ModelLearningSystemType, TaskStatus } from "./../../../types/graphql-global-types";

// ====================================================
// GraphQL query operation: GetLearning
// ====================================================

export interface GetLearning_modelPlan_opsEvalAndLearning_readyForReviewByUserAccount {
  __typename: "UserAccount";
  id: UUID;
  commonName: string;
}

export interface GetLearning_modelPlan_opsEvalAndLearning {
  __typename: "PlanOpsEvalAndLearning";
  id: UUID;
  ccmInvolvment: CcmInvolvmentType[];
  dataNeededForMonitoring: DataForMonitoringType[];
  iddocSupport: boolean | null;
  modelLearningSystems: ModelLearningSystemType[];
  modelLearningSystemsOther: string | null;
  modelLearningSystemsNote: string | null;
  anticipatedChallenges: string | null;
  readyForReviewByUserAccount: GetLearning_modelPlan_opsEvalAndLearning_readyForReviewByUserAccount | null;
  readyForReviewDts: Time | null;
  status: TaskStatus;
}

export interface GetLearning_modelPlan_operationalNeeds {
  __typename: "OperationalNeed";
  id: UUID;
  modifiedDts: Time | null;
}

export interface GetLearning_modelPlan {
  __typename: "ModelPlan";
  id: UUID;
  modelName: string;
  opsEvalAndLearning: GetLearning_modelPlan_opsEvalAndLearning;
  operationalNeeds: GetLearning_modelPlan_operationalNeeds[];
}

export interface GetLearning {
  modelPlan: GetLearning_modelPlan;
}

export interface GetLearningVariables {
  id: UUID;
}
