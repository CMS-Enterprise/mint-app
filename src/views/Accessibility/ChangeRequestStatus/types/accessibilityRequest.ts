/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AccessibilityRequestStatus } from "./../../../../types/graphql-global-types";

// ====================================================
// GraphQL query operation: accessibilityRequest
// ====================================================

export interface accessibilityRequest_accessibilityRequest_statusRecord {
  __typename: "AccessibilityRequestStatusRecord";
  status: AccessibilityRequestStatus;
}

export interface accessibilityRequest_accessibilityRequest {
  __typename: "AccessibilityRequest";
  id: UUID;
  name: string;
  statusRecord: accessibilityRequest_accessibilityRequest_statusRecord;
}

export interface accessibilityRequest {
  accessibilityRequest: accessibilityRequest_accessibilityRequest | null;
}

export interface accessibilityRequestVariables {
  id: UUID;
}
