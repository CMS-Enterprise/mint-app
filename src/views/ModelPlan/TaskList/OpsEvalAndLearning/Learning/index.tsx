import React, { Fragment, useRef } from 'react';
import { useTranslation } from 'react-i18next';
import { Link, useHistory, useParams } from 'react-router-dom';
import { useMutation, useQuery } from '@apollo/client';
import {
  Breadcrumb,
  BreadcrumbBar,
  BreadcrumbLink,
  Button,
  IconArrowBack,
  Label,
  TextInput
} from '@trussworks/react-uswds';
import { Field, FieldArray, Form, Formik, FormikProps } from 'formik';

import AddNote from 'components/AddNote';
import AskAQuestion from 'components/AskAQuestion';
import ITToolsWarning from 'components/ITToolsWarning';
import PageHeading from 'components/PageHeading';
import PageNumber from 'components/PageNumber';
import ReadyForReview from 'components/ReadyForReview';
import AutoSave from 'components/shared/AutoSave';
import CheckboxField from 'components/shared/CheckboxField';
import { ErrorAlert, ErrorAlertMessage } from 'components/shared/ErrorAlert';
import FieldErrorMsg from 'components/shared/FieldErrorMsg';
import FieldGroup from 'components/shared/FieldGroup';
import TextAreaField from 'components/shared/TextAreaField';
import useScrollElement from 'hooks/useScrollElement';
import GetLearning from 'queries/OpsEvalAndLearning/GetLearning';
import {
  GetLearning as GetLearningType,
  GetLearning_modelPlan_opsEvalAndLearning as GetLearningFormType,
  GetLearningVariables
} from 'queries/OpsEvalAndLearning/types/GetLearning';
import { UpdatePlanOpsEvalAndLearningVariables } from 'queries/OpsEvalAndLearning/types/UpdatePlanOpsEvalAndLearning';
import UpdatePlanOpsEvalAndLearning from 'queries/OpsEvalAndLearning/UpdatePlanOpsEvalAndLearning';
import { ModelLearningSystemType } from 'types/graphql-global-types';
import flattenErrors from 'utils/flattenErrors';
import { dirtyInput } from 'utils/formDiff';
import {
  sortOtherEnum,
  translateModelLearningSystemType
} from 'utils/modelPlan';
import sanitizeStatus from 'utils/status';
import { NotFoundPartial } from 'views/NotFound';

import { isCCWInvolvement, renderCurrentPage, renderTotalPages } from '..';

const Learning = () => {
  const { t } = useTranslation('operationsEvaluationAndLearning');
  const { t: h } = useTranslation('draftModelPlan');
  const { modelID } = useParams<{ modelID: string }>();

  // Omitting readyForReviewBy and readyForReviewDts from initialValues and getting submitted through Formik
  type InitialValueType = Omit<
    GetLearningFormType,
    'readyForReviewBy' | 'readyForReviewDts'
  >;

  const formikRef = useRef<FormikProps<InitialValueType>>(null);
  const history = useHistory();

  const { data, loading, error } = useQuery<
    GetLearningType,
    GetLearningVariables
  >(GetLearning, {
    variables: {
      id: modelID
    }
  });

  const {
    id,
    iddocSupport,
    ccmInvolvment,
    modelLearningSystems,
    modelLearningSystemsOther,
    modelLearningSystemsNote,
    anticipatedChallenges,
    readyForReviewBy,
    readyForReviewDts,
    status
  } = data?.modelPlan?.opsEvalAndLearning || ({} as GetLearningFormType);

  const modelName = data?.modelPlan?.modelName || '';

  const itSolutionsStarted: boolean = !!data?.modelPlan.operationalNeeds.find(
    need => need.modifiedDts
  );

  // If redirected from IT Tools, scrolls to the relevant question
  useScrollElement(!loading);

  const [update] = useMutation<UpdatePlanOpsEvalAndLearningVariables>(
    UpdatePlanOpsEvalAndLearning
  );

  const handleFormSubmit = (
    redirect?: 'next' | 'back' | 'task-list' | string
  ) => {
    const dirtyInputs = dirtyInput(
      formikRef?.current?.initialValues,
      formikRef?.current?.values
    );

    if (dirtyInputs.status) {
      dirtyInputs.status = sanitizeStatus(dirtyInputs.status);
    }

    update({
      variables: {
        id,
        changes: dirtyInputs
      }
    })
      .then(response => {
        if (!response?.errors) {
          if (redirect === 'back') {
            history.push(
              `/models/${modelID}/task-list/ops-eval-and-learning/data-sharing`
            );
          } else if (redirect === 'task-list') {
            history.push(`/models/${modelID}/task-list`);
          } else if (redirect) {
            history.push(redirect);
          }
        }
      })
      .catch(errors => {
        formikRef?.current?.setErrors(errors);
      });
  };

  const initialValues: InitialValueType = {
    __typename: 'PlanOpsEvalAndLearning',
    id: id ?? '',
    iddocSupport: iddocSupport ?? null,
    ccmInvolvment: ccmInvolvment ?? [],
    modelLearningSystems: modelLearningSystems ?? [],
    modelLearningSystemsOther: modelLearningSystemsOther ?? '',
    modelLearningSystemsNote: modelLearningSystemsNote ?? '',
    anticipatedChallenges: anticipatedChallenges ?? '',
    status
  };

  if ((!loading && error) || (!loading && !data?.modelPlan)) {
    return <NotFoundPartial />;
  }

  return (
    <>
      <BreadcrumbBar variant="wrap">
        <Breadcrumb>
          <BreadcrumbLink asCustom={Link} to="/">
            <span>{h('home')}</span>
          </BreadcrumbLink>
        </Breadcrumb>
        <Breadcrumb>
          <BreadcrumbLink asCustom={Link} to={`/models/${modelID}/task-list/`}>
            <span>{h('tasklistBreadcrumb')}</span>
          </BreadcrumbLink>
        </Breadcrumb>
        <Breadcrumb current>{t('breadcrumb')}</Breadcrumb>
      </BreadcrumbBar>
      <PageHeading className="margin-top-4 margin-bottom-2">
        {t('heading')}
      </PageHeading>

      <p
        className="margin-top-0 margin-bottom-1 font-body-lg"
        data-testid="model-plan-name"
      >
        {h('for')} {modelName}
      </p>
      <p className="margin-bottom-2 font-body-md line-height-sans-4">
        {h('helpText')}
      </p>

      <AskAQuestion modelID={modelID} />

      <Formik
        initialValues={initialValues}
        onSubmit={() => {
          handleFormSubmit('task-list');
        }}
        enableReinitialize
        innerRef={formikRef}
      >
        {(formikProps: FormikProps<InitialValueType>) => {
          const {
            errors,
            handleSubmit,
            setFieldValue,
            setErrors,
            values
          } = formikProps;
          const flatErrors = flattenErrors(errors);

          return (
            <>
              {Object.keys(errors).length > 0 && (
                <ErrorAlert
                  testId="formik-validation-errors"
                  classNames="margin-top-3"
                  heading={h('checkAndFix')}
                >
                  {Object.keys(flatErrors).map(key => {
                    return (
                      <ErrorAlertMessage
                        key={`Error.${key}`}
                        errorKey={key}
                        message={flatErrors[key]}
                      />
                    );
                  })}
                </ErrorAlert>
              )}

              <Form
                className="tablet:grid-col-6 margin-top-6"
                data-testid="ops-eval-and-learning-learning-form"
                onSubmit={e => {
                  handleSubmit(e);
                }}
              >
                <FieldArray
                  name="modelLearningSystems"
                  render={arrayHelpers => (
                    <>
                      <legend className="usa-label">
                        {t('learningSystem')}
                      </legend>

                      {itSolutionsStarted && (
                        <ITToolsWarning
                          id="ops-eval-and-learning-learning-systems-warning"
                          onClick={() =>
                            handleFormSubmit(
                              `/models/${modelID}/task-list/it-solutions`
                            )
                          }
                        />
                      )}

                      <FieldErrorMsg>
                        {flatErrors.modelLearningSystems}
                      </FieldErrorMsg>

                      {Object.keys(ModelLearningSystemType)
                        .sort(sortOtherEnum)
                        .map(type => {
                          return (
                            <Fragment key={type}>
                              <Field
                                as={CheckboxField}
                                id={`ops-eval-and-learning-learning-systems-${type}`}
                                name="modelLearningSystems"
                                label={translateModelLearningSystemType(type)}
                                value={type}
                                checked={values?.modelLearningSystems.includes(
                                  type as ModelLearningSystemType
                                )}
                                onChange={(
                                  e: React.ChangeEvent<HTMLInputElement>
                                ) => {
                                  if (e.target.checked) {
                                    arrayHelpers.push(e.target.value);
                                  } else {
                                    const idx = values.modelLearningSystems.indexOf(
                                      e.target.value as ModelLearningSystemType
                                    );
                                    arrayHelpers.remove(idx);
                                  }
                                }}
                              />
                              {type === ModelLearningSystemType.OTHER &&
                                values.modelLearningSystems.includes(type) && (
                                  <div className="margin-left-4 margin-top-neg-3">
                                    <Label
                                      htmlFor="ops-eval-and-learning-learning-systems-other"
                                      className="text-normal maxw-none"
                                    >
                                      {h('pleaseSpecify')}
                                    </Label>
                                    <FieldErrorMsg>
                                      {flatErrors.modelLearningSystemsOther}
                                    </FieldErrorMsg>
                                    <Field
                                      as={TextInput}
                                      className="maxw-none"
                                      id="ops-eval-and-learning-learning-systems-other"
                                      maxLength={50}
                                      name="modelLearningSystemsOther"
                                    />
                                  </div>
                                )}
                            </Fragment>
                          );
                        })}
                      <AddNote
                        id="ops-eval-and-learning-learning-systems-note"
                        field="modelLearningSystemsNote"
                      />
                    </>
                  )}
                />

                <FieldGroup
                  scrollElement="anticipatedChallenges"
                  error={!!flatErrors.anticipatedChallenges}
                >
                  <Label htmlFor="ops-eval-and-learning-learning-anticipated-challenges">
                    {t('obstacles')}
                  </Label>
                  <p className="text-base margin-y-1 margin-top-2 line-height-body-4">
                    {t('obstaclesInfo')}
                  </p>
                  <FieldErrorMsg>
                    {flatErrors.anticipatedChallenges}
                  </FieldErrorMsg>
                  <Field
                    as={TextAreaField}
                    className="height-card"
                    error={flatErrors.anticipatedChallenges}
                    id="ops-eval-and-learning-learning-anticipated-challenges"
                    data-testid="ops-eval-and-learning-learning-anticipated-challenges"
                    name="anticipatedChallenges"
                  />
                </FieldGroup>

                <ReadyForReview
                  id="ops-eval-and-learning-learning-status"
                  field="status"
                  sectionName={t('heading')}
                  status={values.status}
                  setFieldValue={setFieldValue}
                  readyForReviewBy={readyForReviewBy}
                  readyForReviewDts={readyForReviewDts}
                />

                <div className="margin-top-6 margin-bottom-3">
                  <Button
                    type="button"
                    className="usa-button usa-button--outline margin-bottom-1"
                    onClick={() => {
                      handleFormSubmit('back');
                    }}
                  >
                    {h('back')}
                  </Button>
                  <Button type="submit" onClick={() => setErrors({})}>
                    {h('saveAndStartNext')}
                  </Button>
                </div>
                <Button
                  type="button"
                  className="usa-button usa-button--unstyled"
                  onClick={() => handleFormSubmit('task-list')}
                >
                  <IconArrowBack className="margin-right-1" aria-hidden />
                  {h('saveAndReturn')}
                </Button>
              </Form>

              {id && (
                <AutoSave
                  values={values}
                  onSave={() => {
                    handleFormSubmit();
                  }}
                  debounceDelay={3000}
                />
              )}
            </>
          );
        }}
      </Formik>
      {data && (
        <PageNumber
          currentPage={renderCurrentPage(
            9,
            iddocSupport,
            isCCWInvolvement(ccmInvolvment)
          )}
          totalPages={renderTotalPages(
            iddocSupport,
            isCCWInvolvement(ccmInvolvment)
          )}
          className="margin-y-6"
        />
      )}
    </>
  );
};

export default Learning;
