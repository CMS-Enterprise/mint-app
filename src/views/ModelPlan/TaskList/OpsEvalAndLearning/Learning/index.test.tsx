import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render, screen, waitFor } from '@testing-library/react';

import GetLearning from 'queries/OpsEvalAndLearning/GetLearning';
import { GetLearning_modelPlan_opsEvalAndLearning as GetLearningType } from 'queries/OpsEvalAndLearning/types/GetLearning';
import { CcmInvolvmentType, TaskStatus } from 'types/graphql-global-types';

import Learning from '.';

const learningMockData: GetLearningType = {
  __typename: 'PlanOpsEvalAndLearning',
  id: '123',
  ccmInvolvment: [CcmInvolvmentType.YES_EVALUATION],
  iddocSupport: true,
  modelLearningSystems: [],
  modelLearningSystemsOther: '',
  modelLearningSystemsNote: '',
  anticipatedChallenges: 'Hard challenges',
  readyForReviewBy: 'ASDF',
  readyForReviewDts: '2022-05-12T15:01:39.190679Z',
  status: TaskStatus.IN_PROGRESS
};

const learningMock = [
  {
    request: {
      query: GetLearning,
      variables: { id: 'ce3405a0-3399-4e3a-88d7-3cfc613d2905' }
    },
    result: {
      data: {
        modelPlan: {
          id: 'ce3405a0-3399-4e3a-88d7-3cfc613d2905',
          modelName: 'My excellent plan that I just initiated',
          opsEvalAndLearning: learningMockData,
          itTools: {
            status: TaskStatus.IN_PROGRESS
          }
        }
      }
    }
  }
];

describe('Model Plan Ops Eval and Learning - Learning', () => {
  it('renders without errors', async () => {
    render(
      <MemoryRouter
        initialEntries={[
          '/models/ce3405a0-3399-4e3a-88d7-3cfc613d2905/task-list/ops-eval-and-learning/learning'
        ]}
      >
        <MockedProvider mocks={learningMock} addTypename={false}>
          <Route path="/models/:modelID/task-list/ops-eval-and-learning/learning">
            <Learning />
          </Route>
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(
        screen.getByTestId('ops-eval-and-learning-learning-form')
      ).toBeInTheDocument();
    });

    await waitFor(() => {
      expect(
        screen.getByTestId(
          'ops-eval-and-learning-learning-anticipated-challenges'
        )
      ).toHaveValue('Hard challenges');
    });
  });

  it('matches snapshot', async () => {
    const { asFragment } = render(
      <MemoryRouter
        initialEntries={[
          '/models/ce3405a0-3399-4e3a-88d7-3cfc613d2905/task-list/ops-eval-and-learning/learning'
        ]}
      >
        <MockedProvider mocks={learningMock} addTypename={false}>
          <Route path="/models/:modelID/task-list/ops-eval-and-learning/learning">
            <Learning />
          </Route>
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(
        screen.getByTestId(
          'ops-eval-and-learning-learning-anticipated-challenges'
        )
      ).toHaveValue('Hard challenges');
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
