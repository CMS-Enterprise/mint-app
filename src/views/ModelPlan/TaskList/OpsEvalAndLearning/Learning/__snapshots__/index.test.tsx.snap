// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model Plan Ops Eval and Learning - Learning matches snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="Breadcrumbs"
    class="usa-breadcrumb usa-breadcrumb--wrap"
  >
    <ol
      class="usa-breadcrumb__list"
    >
      <li
        class="usa-breadcrumb__list-item"
      >
        <a
          class="usa-breadcrumb__link"
          href="/"
        >
          <span>
            Home
          </span>
        </a>
      </li>
      <li
        class="usa-breadcrumb__list-item"
      >
        <a
          class="usa-breadcrumb__link"
          href="/models/ce3405a0-3399-4e3a-88d7-3cfc613d2905/task-list/"
        >
          <span>
            Model Plan task list
          </span>
        </a>
      </li>
      <li
        aria-current="page"
        class="usa-breadcrumb__list-item usa-current"
      >
        Operations, evaluation, and learning
      </li>
    </ol>
  </nav>
  <h1
    aria-live="polite"
    class="mint-h1 margin-top-4 margin-bottom-2"
    tabindex="-1"
  >
    Operations, evaluation, and learning
  </h1>
  <p
    class="margin-top-0 margin-bottom-1 font-body-lg"
    data-testid="model-plan-name"
  >
    for My excellent plan that I just initiated
  </p>
  <p
    class="margin-bottom-2 font-body-md line-height-sans-4"
  >
    If there's a question or field that is not applicable to your model or you don't currently know the answer, you may leave it blank. If you need help, ask a question using the link below.
  </p>
  <div
    class="padding-2 bg-primary-lighter"
  >
    <div
      class="display-flex"
      data-testid="ask-a-question"
    >
      <svg
        class="usa-icon text-primary margin-right-1"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
        />
      </svg>
      <button
        class="usa-button usa-button--unstyled"
        data-testid="ask-a-question-button"
        type="button"
      >
        Ask a question
      </button>
    </div>
  </div>
  <form
    action="#"
    class="desktop:grid-col-6 margin-top-6"
    data-testid="ops-eval-and-learning-learning-form"
  >
    <div
      class="usa-form-group"
      data-scroll="modelLearningSystems"
    >
      <legend
        class="usa-label"
      >
        Will the model have a learning strategy?
      </legend>
      <div
        class="usa-alert usa-alert--warning flex it-solutions-warning"
        data-testid="alert"
      >
        <div
          class="usa-alert__body"
        >
          <p
            class="usa-alert__text"
          >
            <span>
              Changing your answer to this question may also affect your selections in the IT solutions and implementation status tracker.  
              <button
                class="usa-button usa-button usa-button--unstyled"
                data-testid="button"
                id="ops-eval-and-learning-learning-systems-warning"
                type="button"
              >
                Go to IT solutions tracker.
              </button>
            </span>
          </p>
        </div>
      </div>
      <div
        class="mint-checkbox usa-checkbox"
      >
        <input
          class="usa-checkbox__input"
          id="ops-eval-and-learning-learning-systems-EDUCATE_BENEFICIARIES"
          name="modelLearningSystems"
          type="checkbox"
          value="EDUCATE_BENEFICIARIES"
        />
        <label
          class="usa-checkbox__label"
          for="ops-eval-and-learning-learning-systems-EDUCATE_BENEFICIARIES"
        >
          We plan to educate beneficiaries
        </label>
        <p
          class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs"
        />
      </div>
      <div
        class="mint-checkbox usa-checkbox"
      >
        <input
          class="usa-checkbox__input"
          id="ops-eval-and-learning-learning-systems-IT_PLATFORM_CONNECT"
          name="modelLearningSystems"
          type="checkbox"
          value="IT_PLATFORM_CONNECT"
        />
        <label
          class="usa-checkbox__label"
          for="ops-eval-and-learning-learning-systems-IT_PLATFORM_CONNECT"
        >
          We plan to use an IT platform (Connect)
        </label>
        <p
          class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs"
        />
      </div>
      <div
        class="mint-checkbox usa-checkbox"
      >
        <input
          class="usa-checkbox__input"
          id="ops-eval-and-learning-learning-systems-LEARNING_CONTRACTOR"
          name="modelLearningSystems"
          type="checkbox"
          value="LEARNING_CONTRACTOR"
        />
        <label
          class="usa-checkbox__label"
          for="ops-eval-and-learning-learning-systems-LEARNING_CONTRACTOR"
        >
          We plan to have a learning contractor (cross-model or individual)
        </label>
        <p
          class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs"
        />
      </div>
      <div
        class="mint-checkbox usa-checkbox"
      >
        <input
          class="usa-checkbox__input"
          id="ops-eval-and-learning-learning-systems-PARTICIPANT_COLLABORATION"
          name="modelLearningSystems"
          type="checkbox"
          value="PARTICIPANT_COLLABORATION"
        />
        <label
          class="usa-checkbox__label"
          for="ops-eval-and-learning-learning-systems-PARTICIPANT_COLLABORATION"
        >
          We plan to enable participant-to-participant collaboration
        </label>
        <p
          class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs"
        />
      </div>
      <div
        class="mint-checkbox usa-checkbox"
      >
        <input
          class="usa-checkbox__input"
          id="ops-eval-and-learning-learning-systems-OTHER"
          name="modelLearningSystems"
          type="checkbox"
          value="OTHER"
        />
        <label
          class="usa-checkbox__label"
          for="ops-eval-and-learning-learning-systems-OTHER"
        >
          Other
        </label>
        <p
          class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs"
        />
      </div>
      <div
        class="mint-checkbox usa-checkbox"
      >
        <input
          class="usa-checkbox__input"
          id="ops-eval-and-learning-learning-systems-NO_LEARNING_SYSTEM"
          name="modelLearningSystems"
          type="checkbox"
          value="NO_LEARNING_SYSTEM"
        />
        <label
          class="usa-checkbox__label"
          for="ops-eval-and-learning-learning-systems-NO_LEARNING_SYSTEM"
        >
          No, we will not have a learning strategy
        </label>
        <p
          class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs"
        />
      </div>
      <div
        class="margin-top-4 margin-bottom-8"
      >
        <button
          class="usa-button usa-button usa-button--unstyled"
          data-testid="add-note-toggle"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="usa-icon margin-right-1"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          Add an additional note
        </button>
      </div>
    </div>
    <div
      class="usa-form-group"
      data-scroll="anticipatedChallenges"
    >
      <label
        class="usa-label"
        data-testid="label"
        for="ops-eval-and-learning-learning-anticipated-challenges"
      >
        What challenges do you anticipate during Model design and implementation?
      </label>
      <p
        class="text-base margin-y-1 margin-top-2 line-height-body-4"
      >
        Please list and known ’unknowns,’ that is, are there policy decisions that you are aware of that are still pending or are subject to change? If so, please list to the best of your ability.
      </p>
      <textarea
        aria-labelledby="label-ops-eval-and-learning-learning-anticipated-challenges hint-ops-eval-and-learning-learning-anticipated-challenges"
        class="height-card usa-textarea"
        data-testid="ops-eval-and-learning-learning-anticipated-challenges"
        id="ops-eval-and-learning-learning-anticipated-challenges"
        name="anticipatedChallenges"
      >
        Hard challenges
      </textarea>
    </div>
    <div
      class="usa-form-group margin-top-8 margin-bottom-3"
    >
      <div
        class="usa-summary-box bg-white border-base-light padding-2"
        data-testid="summary-box"
      >
        <div
          class="usa-summary-box__body"
        >
          <h3
            class="usa-summary-box__heading"
          />
          <div
            class="usa-summary-box__text"
          >
            <p
              class="margin-0"
            >
              Model Plan status
            </p>
            <div
              class="mint-checkbox usa-checkbox"
            >
              <input
                class="usa-checkbox__input"
                data-testid="ops-eval-and-learning-learning-status"
                id="ops-eval-and-learning-learning-status"
                name="status"
                type="checkbox"
                value="IN_PROGRESS"
              />
              <label
                class="usa-checkbox__label"
                for="ops-eval-and-learning-learning-status"
              >
                This section of the Model Plan (Operations, evaluation, and learning) is ready for review.
              </label>
              <p
                class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="margin-top-6 margin-bottom-3"
    >
      <button
        class="usa-button usa-button usa-button--outline margin-bottom-1"
        data-testid="button"
        type="button"
      >
        Back
      </button>
      <button
        class="usa-button"
        data-testid="button"
        type="submit"
      >
        Save and start next Model Plan section
      </button>
    </div>
    <button
      class="usa-button usa-button usa-button--unstyled"
      data-testid="button"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="usa-icon margin-right-1"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
      Save and return to task list
    </button>
  </form>
  <div
    class="mint-page-number margin-y-6"
  >
    <span
      class="mint-page-number__page-num"
    >
      Page 9 of 9
    </span>
  </div>
</DocumentFragment>
`;
